apps/io-services-cms-webapp/src/webservice/controllers/regenerate-service-keys.ts

```diff
@@ -55,12 +58,35 @@ type RegenerateServiceKeysHandler = (
 
 interface Dependencies {
   apimService: ApimUtils.ApimService;
+  fsmLifecycleClient: ServiceLifecycle.FsmClient;
   telemetryClient: TelemetryClient;
 }
 
+const checkService =
+  (fsmLifecycleClient: ServiceLifecycle.FsmClient) =>
+  (serviceId: NonEmptyString): TE.TaskEither<ErrorResponseTypes, void> =>
+    pipe(
+      serviceId,
+      fsmLifecycleClient.getStore().fetch,
+      TE.mapLeft((err) => ResponseErrorInternal(err.message)),
+      TE.chainW(
+        flow(
+          O.filter((service) => service.fsm.state !== "deleted"),
+          O.map(() => void 0),
+          TE.fromOption(() =>
+            ResponseErrorNotFound(
+              "Not found",
+              `no item with id ${serviceId} found`,
+            ),
+          ),
+        ),
+      ),
+    );
```

Comment:
This function can be moved to a different module and imported both in here and in `get-service-key` controller!